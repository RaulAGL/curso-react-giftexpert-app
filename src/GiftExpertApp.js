

import React, {useState} from 'react'
import AddCategory from "./components/AddCategory";
import GiftGrid from "./components/GiftGrid";

const GifExpertApp = ({ defaultcategories = [] } ) => {

    const [categories,setCategories] = useState( defaultcategories );

    return (
        <>
            <h2>GiftExpertApp</h2>
            <hr/>
            <AddCategory setCategories={setCategories} />
            {
                categories.map( (category) => {
                    return <GiftGrid key={category} category={category} />
                })
            }
        </>
    );
}


export default GifExpertApp;
