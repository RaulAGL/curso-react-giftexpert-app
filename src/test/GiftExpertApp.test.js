
import GiftExpertApp from '../GiftExpertApp';
import React from 'react';
import {shallow} from 'enzyme';

describe('Probar funcionalidad en <GiftExpertApp />', () => {


    test('probar snapshot de GiftExpertApp', () => {

        const wrapper = shallow( <GiftExpertApp /> );

        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de mostrar una lista de categories', () => {

        const categories = ['the last of us', 'the last of us part II'];
        const wrapper = shallow( <GiftExpertApp defaultcategories={ categories } />);

        expect( wrapper ).toMatchSnapshot();

        expect( wrapper.find('GiftGrid').length ).toBe( categories.length );
    });
});
