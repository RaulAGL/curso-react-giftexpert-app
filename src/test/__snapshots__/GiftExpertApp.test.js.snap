// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar funcionalidad en <GiftExpertApp /> debe de mostrar una lista de categories 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <GiftGrid
    category="the last of us"
    key="the last of us"
  />
  <GiftGrid
    category="the last of us part II"
    key="the last of us part II"
  />
</Fragment>
`;

exports[`Probar funcionalidad en <GiftExpertApp /> probar snapshot de GiftExpertApp 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
</Fragment>
`;
